Java code for Binary Search in 2D arrays

priyanshuhaldar007 opened the pull request:
Added a java code for doing a binary search in partially sorted 2D arrays. Please accept this PR

Saikat2407 reviewed (changes requested):
Add this file under java folder of searching folder

priyanshuhaldar007 commented:
> Add this file under java folder of searching folder

Added the file to its correct directory and also added one more question in the Leetcode section

Saikat2407 reviewed (approved):
**Approved**
